### DISCUSSION TITLE
Increase default.replication.factor to topics in-place?

### DISCUSSION THREAD

**Discussion by jslusher** [strimzi/strimzi-kafka-operator#4348] [General]
I have a Strimzi (version `0.16.2`) cluster and I'm running some debezium connectors in it. Originally I had set `default.replication.factor` to 2 on the `Kafka` CRD and I created a handful of connectors using the kafka-connect API (not using Strimzi objects). I then went back and set `default.replication.factor` to 3 on the `Kafka` CRD. I see that the setting is in place if I look at the nodes themselves. I expected that would increase the number of replicas for the connector topics to 3 but I still only see there are 2 replicas. Am I misunderstanding how a connector interacts with its topics or should that have increased the replicas on the connector's topics?

**Comment by scholzj** [comment]
I think this setting applies only to newly created topics using the _auto-creation_. It does not impact already existing topics. For existing topics, you would need to use the `kafka-reassign-partitions.sh` tool.

**Reply by jslusher** [reply]
Ah I see. That makes sense. I'll look into that option. Thanks!
